src/lib/linux-api/src/signal.rs

```diff
@@ -942,10 +947,39 @@ pub enum SignalHandler {
 #[allow(non_camel_case_types)]
 pub type linux_sigaction = bindings::linux_sigaction;
 
+/// Restorer suitable for use with `sigaction`.
+///
+/// Normally libc's implementation of the `sigaction` function injects a similar
+/// restorer function.
+///
+/// # Safety
+///
+/// This function is only intended for use as a `restorer` in `sigaction`.
+/// Do not call this function directly.
```

Comment:
Can you expand the documentation a bit to explain what this does and why/when it's used? For example what does the sigaction restorer do, and why is it marked as a `naked_function::naked`?

Reply:
done. Mostly referenced and pasted `sigreturn(2)`.